### PR TITLE
Bug fix 3.3/pathuniqueness in bfs

### PR DESCRIPTION
**Pull request by mchacki**
Fixes issue #5941 path uniqueness was not validated if breath first search was executed.

Jenkins:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/888/

### CODE REVIEW

**Comment by mchacki** [comment]
Updated Jekins:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/890/
The old one suffered from jenkins issues, not PR related.

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,10 @@
 v3.3.15 (XXXX-XX-XX)
 --------------------
 
+* fixed issue #5941 if using breath first search in traversals uniqueness checks
```
Comment (Simran-B) [inline CHANGELOG]: breath -> breadth

File: arangod/Graph/BreadthFirstEnumerator.cpp [hunk 2]
```diff
@@ -221,3 +235,37 @@ arangodb::aql::AqlValue BreadthFirstEnumerator::pathToAqlValue(
   result.close();
   return arangodb::aql::AqlValue(result.slice());
 }
+
+bool BreadthFirstEnumerator::pathContainsVertex(size_t index, StringRef vertex) const {
+  while (true) {
+    TRI_ASSERT(index < _schreier.size());
+    auto const& step = _schreier[index];
+    // Massive logic error, only valid pointers should inserted in _schreier
```
Comment (Simran-B) [inline arangod/Graph/BreadthFirstEnumerator.cpp]: should be inserted into?

File: arangod/Graph/BreadthFirstEnumerator.cpp [hunk 3]
```diff
@@ -221,3 +235,37 @@ arangodb::aql::AqlValue BreadthFirstEnumerator::pathToAqlValue(
...
+    TRI_ASSERT(step != nullptr);
+    if (step->vertex == vertex) {
+      // We have the given vertex on this path
+      return true;
+    }
+    if (index == 0) {
+      //We have checked the compelte path
```
Comment (Simran-B) [inline arangod/Graph/BreadthFirstEnumerator.cpp]: -> complete

File: arangod/Graph/BreadthFirstEnumerator.cpp [hunk 4]
```diff
@@ -221,3 +235,37 @@ arangodb::aql::AqlValue BreadthFirstEnumerator::pathToAqlValue(
...
+    index = step->sourceIdx;
+  }
+  return false;
+}
+
+bool BreadthFirstEnumerator::pathContainsEdge(size_t index, graph::EdgeDocumentToken const& edge) const {
+  while (index != 0) {
+    TRI_ASSERT(index < _schreier.size());
+    auto const& step = _schreier[index];
+    // Massive logic error, only valid pointers should inserted in _schreier
```
Comment (Simran-B) [inline arangod/Graph/BreadthFirstEnumerator.cpp]: s.o.

**Pull request merged**